ultralytics/engine/results.py

```diff
@@ -327,6 +334,100 @@ def verbose(self):
                 log_string += f"{n} {self.names[int(c)]}{'s' * (n > 1)}, "
         return log_string
 
+    def asdict(self, py_typed: bool = True) -> dict:
+        """
+        Convert the detection results for any task into a dictionary format.
+
+        Args:
+            py_typed (bool): Whether to convert values in the dictionary to Python types; used to save as YAML.
+
+        Returns:
+            dict: A dictionary containing results data.
+        """
+        # Check task data
+        box, mask, probs, kp, obb = [getattr(self, t) is not None for t in self._keys]
+        r = self.cpu().numpy()
+        lbl_idx = [r.probs.top1] if probs else (r.obb.cls if obb else r.boxes.cls)
+        all_conf = r.probs.top1conf if probs else (r.obb.conf if obb else r.boxes.conf)
+        all_ids = r.boxes.id if box else r.obb.id
```

Comment:
Script is going to crash on the classification task since boxes are empty in that case

```suggestion
        all_ids = r.boxes.id if box else (r.obb.id if obb else None)
        if isinstance(all_conf, np.float32):
            all_conf = [all_conf]
```

Reply:
You're right! Thank you for catching that. I went a slightly different route that uses less code (although breaks the pattern a bit). I also noticed that there was an issue with `all_conf` while testing and made an update to this as well.